domain/user/service/service.go

```diff
@@ -159,24 +207,46 @@ func (s *Service) AddUser(ctx context.Context, user user.User) error {
 // - internal/auth.ErrPasswordDestroyed: If the supplied password has already
 // been destroyed.
 // - internal/auth.ErrPasswordNotValid: If the password supplied is not valid.
-func (s *Service) AddUserWithPassword(ctx context.Context, user user.User, password auth.Password) error {
+func (s *Service) AddUserWithPassword(ctx context.Context, usr user.User, creatorUUID user.UUID, password auth.Password) error {
 	defer password.Destroy()
-	if err := ValidateUsername(user.Name); err != nil {
-		return fmt.Errorf("username %q: %w", user.Name, err)
+	// Validate user name
+	if err := ValidateUsername(usr.Name); err != nil {
+		return fmt.Errorf("username %q: %w", usr.Name, err)
+	}
+
+	// Validate creator UUID
+	if err := creatorUUID.Validate(); err != nil {
+		return fmt.Errorf("creator uuid %q: %w", creatorUUID, err)
 	}
 
+	// Generate a salt for the password.
 	salt, err := auth.NewSalt()
 	if err != nil {
-		return fmt.Errorf("adding user %q, generating password salt: %w", user.Name, err)
+		return fmt.Errorf("adding user %q, generating password salt: %w", usr.Name, err)
 	}
 
+	// Hash the password.
 	pwHash, err := auth.HashPassword(password, salt)
 	if err != nil {
-		return fmt.Errorf("adding user %q, hashing password: %w", user.Name, err)
+		return fmt.Errorf("adding user %q, hashing password: %w", usr.Name, err)
+	}
+
+	// Get creator by UUID
```

Comment:
Small nit. I think we should shift the finding of the creator user to up above the password having. May as well make sure all of our pre conditions exist before working on the crypto.

Reply:
good catch!